packages/status-js/src/utils/encode-url-data.ts

```diff
@@ -0,0 +1,98 @@
+import { base64url } from '@scure/base'
+import { sha256 } from 'ethereum-cryptography/sha256'
+import {
+  concatBytes,
+  utf8ToBytes as toBytes,
+} from 'ethereum-cryptography/utils'
+import { brotliCompressSync, brotliDecompressSync } from 'node:zlib'
+
+import { Channel, Community, URLData, User } from '../protos/url-data_pb'
+
+import type { PlainMessage } from '@bufbuild/protobuf'
+
+export function encodeCommunityUrlData(
+  data: PlainMessage<Community>,
+  publicKey: string
+): string {
+  return encodeUrlData(new Community(data).toBinary(), publicKey)
+}
+
+export function decodeCommunityUrlData(
+  data: string,
+  publicKey: string
+): PlainMessage<Community> {
+  const deserialized = decodeUrlData(data, publicKey)
+  const community = Community.fromBinary(deserialized.content)
+
+  // note: PlainMessage<T> type does not ensure returning of only own properties
+  return { ...community }
+}
+
+export function encodeChannelUrlData(
+  data: PlainMessage<Channel>,
+  publicKey: string
+): string {
+  return encodeUrlData(new Channel(data).toBinary(), publicKey)
+}
+
+export function decodeChannelUrlData(
+  data: string,
+  publicKey: string
+): PlainMessage<Channel> {
+  const deserialized = decodeUrlData(data, publicKey)
+  const channel = Channel.fromBinary(deserialized.content)
+
+  return { ...channel }
+}
+
+export function encodeUserUrlData(data: PlainMessage<User>, publicKey: string) {
+  return encodeUrlData(new User(data).toBinary(), publicKey)
+}
+
+export function decodeUserUrlData(data: string, publicKey: string) {
+  const deserialized = decodeUrlData(data, publicKey)
+  const user = User.fromBinary(deserialized.content)
+
+  return { ...user }
+}
+
+function encodeUrlData(data: Uint8Array, publicKey: string): string {
+  const serialized = new URLData({
+    content: data,
+    checksum: getChecksum(data, publicKey),
+  }).toBinary()
+  const compressed = brotliCompressSync(serialized)
+  // todo?!: remove padding
```

Comment:
It's the `==` that can be at the end of encoded results, and actual savings being max 2 chars.